src/mono/wasm/runtime/loader/icu.ts

```diff
@@ -47,7 +47,7 @@ export function init_globalization() {
 
 export function getIcuResourceName(config: MonoConfig): string | null {
     if (config.resources?.icu && config.globalizationMode != GlobalizationMode.Invariant) {
-        const culture = config.applicationCulture || (ENVIRONMENT_IS_WEB ? (navigator.languages && navigator.languages[0]) : Intl.DateTimeFormat().resolvedOptions().locale);
+        const culture = config.applicationCulture || (ENVIRONMENT_IS_WEB ? (globalThis.navigator && globalThis.navigator.languages && globalThis.navigator.languages[0]) : Intl.DateTimeFormat().resolvedOptions().locale);
```

Comment:
is navigator.languages correct in a worker? we may need to remote it from the main thread to the sidecar eventually if not.

Reply:
good point

Reply:
I will put TODO, because this is not sidecar PR yet, just prep